server/services/courseService.ts

```diff
@@ -45,6 +48,29 @@ export default class CourseService {
     return courseClient.findOfferings(courseId)
   }
 
+  async getParticipation(
+    token: Express.User['token'],
+    courseParticipationId: CourseParticipation['id'],
+  ): Promise<CourseParticipation | null> {
+    const courseClient = this.courseClientBuilder(token)
+
+    try {
+      const courseParticipation = await courseClient.findParticipation(courseParticipationId)
+
+      return courseParticipation
+    } catch (error) {
+      const knownError = error as ResponseError
+
+      if (knownError.status === 404) {
+        return null
+      }
+
+      throw createError(knownError.status || 500, knownError, {
+        userMessage: `Error fetching course participation with ID ${courseParticipationId}.`,
+      })
+    }
```

Comment:
/do we need this error handling when we're working with our internal API (other internal API service/client methods don't do this)?

Reply:
Same as the tests really, I think it's worth adding some error handling.  I have actually updated this slightly in an upcoming PR/commit too as it's misusing `createError` and also think its' worth throwing an error if not found, instead of just `null`.

Reply:
Yeah maybe - I guess the reason we haven't done this so far is that we'd not expect our UI to show any data for the stuff we own if it doesn't exist in our API, so there shouldn't be 404s unless something's been deleted or the URL has been manually edited

Thinking about it, it probably does make more sense for participations since they're deletable, vs courses and referrals which are not (so you'd need to manually edit a URL or a form to try to get/post/put to a non-existent entity). We might want to add similar for `updateParticipation` in that case?